widgetssdk/src/main/java/com/glia/widgets/core/engagement/GliaEngagementStateRepository.java

```diff
@@ -99,12 +100,16 @@ private EngagementStateEvent mapToEngagementStateChangeEvent(
             EngagementState engagementState,
             @Nullable Operator operator
     ) {
-        if (engagementState == null) {
+        if (engagementState == null && isOngoingEngagement) {
+            isOngoingEngagement = false;
             return new EngagementStateEvent.EngagementEndedEvent();
+        } else if (engagementState == null) {
+            return new EngagementStateEvent.NoEngagementEvent();
         } else if (engagementState.getVisitorStatus() == EngagementState.VisitorStatus.TRANSFERRING) {
             return new EngagementStateEvent.EngagementTransferringEvent();
         } else {
             if (operator == null) {
+                isOngoingEngagement = true;
```

Comment:
I am not sure that `isOngoingEngagement = true;` should only be set `if (operator == null)`. Maybe it should be set at line number 111, for the whole else block?

Reply:
This looks unreasonable, but `if (operator == null)` triggers `EngagementOperatorConnectedEvent` that seems to be the most logical place to mark engagement as started